### PR TITLE
CLOUD-411 Add securityContext field

### PR DESCRIPTION
**Pull request by nonemax**

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | passed |
| init-deploy | passed |
| scaling | passed |
| recreate | passed |
| scaling-proxysql | passed |
| operator-self-healing | passed |
| storage | passed |
| demand-backup | failed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |

File: pkg/apis/pxc/v1/pxc_types.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ type PerconaXtraDBClusterSpec struct {
 	Backup                *PXCScheduledBackup                  `json:"backup,omitempty"`
 	UpdateStrategy        appsv1.StatefulSetUpdateStrategyType `json:"updateStrategy,omitempty"`
 	AllowUnsafeConfig     bool                                 `json:"allowUnsafeConfigurations,omitempty"`
+	SecurityContext       *corev1.PodSecurityContext           `json:"securityContext,omitempty"`
```
Comment (delgod) [inline pkg/apis/pxc/v1/pxc_types.go]: I think it should be possible to set on the level of PXC, ProxySQL, PMM, and storage level of Backup

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | failed |
| scaling | failed |
| self-healing | failed |
| init-deploy | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| init-deploy | passed |
| self-healing | passed |
| scaling | passed |
| scaling-proxysql | passed |
| storage | passed |
| operator-self-healing | passed |
| limits | passed |
| demand-backup | failed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |

File: pkg/apis/pxc/v1/pxc_backup_types.go [hunk 2]
```diff
@@ -26,8 +27,9 @@ type PerconaXtraDBClusterBackup struct {
 }
 
 type PXCBackupSpec struct {
-	PXCCluster  string `json:"pxcCluster"`
-	StorageName string `json:"storageName,omitempty"`
+	PXCCluster      string                     `json:"pxcCluster"`
+	StorageName     string                     `json:"storageName,omitempty"`
+	SecurityContext *corev1.PodSecurityContext `json:"securityContext,omitempty"`
```
Comment (delgod) [inline pkg/apis/pxc/v1/pxc_backup_types.go]: as usual it should be part of storage config

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

File: pkg/apis/pxc/v1/pxc_types.go [hunk 3]
```diff
@@ -35,6 +35,7 @@ type PXCScheduledBackup struct {
 	Storages           map[string]*BackupStorageSpec `json:"storages,omitempty"`
 	ServiceAccountName string                        `json:"serviceAccountName,omitempty"`
 	Resources          *PodResources                 `json:"resources,omitempty"`
+	SecurityContext    *corev1.PodSecurityContext    `json:"securityContext,omitempty"`
```
Comment (delgod) [inline pkg/apis/pxc/v1/pxc_types.go]: as usual it should be part of storage config

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| init-deploy | passed |
| self-healing | passed |
| scaling | passed |
| scaling-proxysql | passed |
| storage | passed |
| operator-self-healing | passed |
| demand-backup | failed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |

File: pkg/pxc/backup/job.go [hunk 4]
```diff
@@ -30,20 +30,11 @@ func (*Backup) Job(cr *api.PerconaXtraDBClusterBackup) *batchv1.Job {
...
-
+func (bcp *Backup) JobSpec(spec api.PXCBackupSpec, cluster api.PerconaXtraDBClusterSpec) batchv1.JobSpec {
 	return batchv1.JobSpec{
 		Template: corev1.PodTemplateSpec{
 			Spec: corev1.PodSpec{
-				SecurityContext: &corev1.PodSecurityContext{
-					SupplementalGroups: []int64{1001},
-					FSGroup:            fsgroup,
-				},
+				SecurityContext:  cluster.Backup.Storages[spec.StorageName].SecurityContext,
```
Comment (delgod) [inline pkg/pxc/backup/job.go]: the same for recovery and scheduled backups

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | passed |
| scaling | passed |
| init-deploy | passed |
| recreate | passed |
| operator-self-healing | passed |
| scaling-proxysql | passed |
| storage | passed |
| limits | passed |
| demand-backup | failed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |

File: pkg/pxc/backup/restore.go [hunk 5]
```diff
@@ -215,6 +215,7 @@ func S3RestoreJob(cr *api.PerconaXtraDBClusterRestore, bcp *api.PerconaXtraDBClu
 			Template: corev1.PodTemplateSpec{
 				Spec: corev1.PodSpec{
 					ImagePullSecrets: cluster.Backup.ImagePullSecrets,
+					SecurityContext:  cluster.Backup.Storages[bcp.Status.StorageName].SecurityContext,
```
Comment (delgod) [inline pkg/pxc/backup/restore.go]: setSecurityContext function doesn't set SecurityContext for all storages

Comment (delgod) [inline pkg/pxc/backup/restore.go]: SecurityContext should be taken from PXC config, not a storage

**Comment by it-percona** [comment]
[CLA assistant check](https://cla-assistant.percona.com:443/percona/percona-xtradb-cluster-operator?pullRequest=266) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.percona.com:443/percona/percona-xtradb-cluster-operator?pullRequest=266) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: nonemax<br/>:x: blez<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.percona.com:443/check/percona/percona-xtradb-cluster-operator?pullRequest=266) it.</sub>

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| init-deploy | failed |
| recreate | passed |
| self-healing | failed |
| scaling-proxysql | passed |
| demand-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | failed |
| recreate | passed |
| demand-backup | failed |
| scaling | passed |
| init-deploy | failed |
| scaling-proxysql | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | failed |
| self-healing | failed |
| scaling | passed |
| recreate | passed |
| scaling-proxysql | passed |
| demand-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | passed |
| recreate | passed |
| demand-backup | failed |
| scaling | passed |
| init-deploy | failed |
| scaling-proxysql | passed |
| operator-self-healing | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| init-deploy | failed |
| self-healing | failed |
| recreate | failed |
| scaling-proxysql | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | failed |
| scaling | passed |
| init-deploy | failed |
| recreate | passed |
| scaling-proxysql | passed |
| demand-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| self-healing | failed |
| init-deploy | failed |
| scaling | passed |
| scaling-proxysql | passed |
| demand-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | passed |
| recreate | passed |
| scaling | passed |
| big-data | failed |
| init-deploy | failed |
| scaling-proxysql | passed |
| demand-backup | failed |
| self-healing-advanced | failed |
| upgrade | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| self-healing | passed |
| big-data | passed |
| recreate | passed |
| init-deploy | failed |
| scaling-proxysql | passed |
| demand-backup | failed |
| upgrade | failed |
| self-healing-advanced | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| recreate | passed |
| self-healing | passed |
| big-data | passed |
| init-deploy | failed |
| scaling-proxysql | passed |
| self-healing-advanced | passed |
| demand-backup | failed |
| upgrade | failed |
| operator-self-healing | passed |
| one-pod | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| scaling | passed |
| self-healing | passed |
| big-data | passed |
| recreate | passed |
| scaling-proxysql | passed |
| storage | failed |
| demand-backup | failed |
| upgrade | failed |
| self-healing-advanced | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | failed |
| recreate | failed |
| scaling | passed |
| self-healing | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| storage | failed |
| upgrade | failed |
| self-healing-advanced | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | passed |
| big-data | passed |
| init-deploy | passed |
| scaling | passed |
| recreate | passed |
| scaling-proxysql | passed |
| storage | failed |
| self-healing-advanced | failed |
| demand-backup | failed |
| upgrade | failed |

File: deploy/cr.yaml [hunk 6]
```diff
@@ -74,7 +74,7 @@ spec:
             storage: 6Gi
     gracePeriod: 600
   proxysql:
-    enabled: true
+    enabled: false
```
Comment (delgod) [inline deploy/cr.yaml]: typo

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| recreate | passed |
| init-deploy | passed |
| self-healing | passed |
| big-data | passed |
| scaling-proxysql | passed |
| storage | failed |
| demand-backup | failed |
| self-healing-advanced | failed |
| upgrade | passed |
| upgrade-consistency | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| self-healing | passed |
| recreate | passed |
| big-data | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| storage | passed |
| demand-backup | failed |
| self-healing-advanced | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| limits | passed |
| operator-self-healing | passed |
| monitoring | failed |
| one-pod | passed |
| auto-tuning | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| init-deploy | passed |
| big-data | passed |
| recreate | passed |
| self-healing | passed |
| storage | passed |
| scaling-proxysql | passed |
| demand-backup | failed |
| limits | passed |
| monitoring | failed |
| upgrade | passed |
| upgrade-consistency | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| one-pod | passed |
| auto-tuning | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| big-data | passed |
| recreate | passed |
| self-healing | passed |
| scaling | passed |
| scaling-proxysql | passed |
| storage | passed |
| demand-backup | failed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| one-pod | passed |
| auto-tuning | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| limits | passed |
| monitoring | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| self-healing | passed |
| init-deploy | failed |
| big-data | passed |
| scaling | passed |
| scaling-proxysql | passed |
| self-healing-advanced | failed |
| demand-backup | failed |
| upgrade | passed |
| upgrade-consistency | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| self-healing | passed |
| recreate | passed |
| scaling | passed |
| scaling-proxysql | passed |
| big-data | failed |
| upgrade | passed |
| upgrade-consistency | passed |
| storage | failed |
| self-healing-advanced | failed |
| demand-backup | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | passed |
| big-data | passed |
| scaling | passed |
| recreate | passed |
| init-deploy | failed |
| scaling-proxysql | passed |
| self-healing-advanced | failed |
| upgrade | passed |
| demand-backup | failed |
| upgrade-consistency | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | failed |
| self-healing | passed |
| recreate | passed |
| init-deploy | failed |
| scaling | passed |
| scaling-proxysql | passed |
| self-healing-advanced | passed |
| demand-backup | passed |
| upgrade | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| one-pod | passed |
| scheduled-backup | failed |
| auto-tuning | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| self-healing | passed |
| scaling | passed |
| init-deploy | passed |
| big-data | passed |
| scaling-proxysql | passed |
| storage | passed |
| demand-backup | passed |
| self-healing-advanced | passed |
| limits | passed |
| upgrade | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| monitoring | passed |
| one-pod | passed |
| monitoring-2-0 | passed |
| scheduled-backup | failed |
| auto-tuning | passed |
| affinity | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| self-healing | passed |
| init-deploy | passed |
| scaling | passed |
| big-data | passed |
| recreate | passed |
| scaling-proxysql | passed |
| self-healing-advanced | failed |
| storage | failed |
| demand-backup | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| scheduled-backup | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| self-healing | passed |
| recreate | passed |
| big-data | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| storage | passed |
| self-healing-advanced | failed |
| upgrade | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| limits | passed |
| monitoring | passed |
| scheduled-backup | passed |
| monitoring-2-0 | passed |
| affinity | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| init-deploy | passed |
| recreate | passed |
| self-healing | passed |
| scaling | passed |
| scaling-proxysql | passed |
| storage | passed |
| self-healing-advanced | failed |
| demand-backup | passed |
| upgrade | passed |
| upgrade-consistency | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| scheduled-backup | passed |

**Pull request merged**